sdk/loadtestservice/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/LoadTestRunAsyncClient.java

```diff
@@ -0,0 +1,1235 @@
+     *     status(ACCEPTED/NOTSTARTED/PROVISIONING/PROVISIONED/CONFIGURING/CONFIGURED/EXECUTING/EXECUTED/DEPROVISIONING/DEPROVISIONED/DONE/CANCELLING/CANCELLED/FAILED/VALIDATION_SUCCESS/VALIDATION_FAILURE).
+     */
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public PollerFlux<BinaryData, BinaryData> beginStartTestRun(
+            String testRunId, BinaryData body, int refreshTime, RequestOptions testRunRequestOptions) {
+        return new PollerFlux<>(
+                Duration.ofSeconds(refreshTime),
+                (context) -> {
+                    Mono<BinaryData> testRunMono =
+                            createOrUpdateTestRunWithResponse(testRunId, body, testRunRequestOptions)
+                                    .flatMap(FluxUtil::toMono);
+                    return testRunMono;
+                },
+                (context) -> {
+                    Mono<BinaryData> testRunMono = getTestRunWithResponse(testRunId, null).flatMap(FluxUtil::toMono);
+                    return testRunMono.flatMap(
+                            testRunBinary -> {
+                                String status;
+                                JsonNode testRun;
+                                try {
+                                    testRun = new ObjectMapper().readTree(testRunBinary.toString());
+                                    status = testRun.get("status").asText();
+                                } catch (JsonProcessingException e) {
+                                    return Mono.error(
+                                            new RuntimeException(
+                                                    "Encountered exception while retriving test run status"));
+                                }
+                                LongRunningOperationStatus lroStatus;
+                                switch (status) {
+                                    case "NOTSTARTED":
+                                        lroStatus = LongRunningOperationStatus.NOT_STARTED;
+                                        break;
+                                    case "DONE":
+                                        lroStatus = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                                        break;
+                                    case "FAILED":
+                                        lroStatus = LongRunningOperationStatus.FAILED;
+                                        break;
+                                    case "CANCELLED":
+                                        lroStatus = LongRunningOperationStatus.USER_CANCELLED;
+                                        break;
+                                    default:
+                                        lroStatus = LongRunningOperationStatus.IN_PROGRESS;
+                                        break;
+                                }
+                                return Mono.just(new PollResponse<>(lroStatus, BinaryData.fromString(status)));
```

Comment:
I am not sure, but should the value to `PollResponse` the payload `testRunBinary`, not only its "status"?

Reply:
Sending entire `testRunBinary` might be bit heavy for the customer, as they might need to do more parsing to get the test status. Currently, they can get the status directly and if they require more info like `testRunBinary` info or the `metrics`, customer can use GET APIs after subscribing to the poller. What do you think?

Reply:
Yes. This depends on how we expect user using the API.

My viewpoint is similar to the beginUpload API, that LongRunningOperationStatus already captures the status of LRO. And the status as enum or string appears to be redundant.

Reply:
Yes, this makes sense. This can also be discussed in arch board meet

Reply:
Also, BinaryData status vs enums for LRO response need to be decided

Reply:
Changed response to `testRunBinary` for now